nuxeo-drive-client/nxdrive/engine/username_resolver.py

```diff
@@ -0,0 +1,41 @@
+'''
+Created on Jul 13, 2016
+
+@author: mkeshava
+'''
+from nxdrive.engine.workers import PollWorker
+from datetime import timedelta, datetime
+from nxdrive.logging_config import get_logger
+
+
+USERNAME_UPDATE_INTERVAL = timedelta(days=1).total_seconds()
+log = get_logger(__name__)
+
+
+class UserNameResolver(PollWorker):
```

Comment:
I think all the methods that handle this, from the "engine" should be moved into this class.. better separation of concerns.
As the engine already holds a reference to this class, it can use it to deal with all aspects of managing users' name.
Just inject it with dependencies it needs, e.g. the REST client, DAO, etc.